tasks/argocd/src/main/java/com/walmartlabs/concord/plugins/argocd/ArgoCdClient.java

```diff
@@ -135,12 +137,33 @@ public V1alpha1Application waitForSync(String app, String resourceVersion, Durat
                 .setConfig(RequestConfig.custom().
                         setSocketTimeout((int) waitTimeout.toMillis()).build())
                 .build();
-        CloseableHttpResponse response = client.getHttpClient().execute(request);
 
+        CloseableHttpResponse response = client.getHttpClient().execute(request);
         BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(response.getEntity().getContent()));
-        String line;
+        String line = null;
+        int count = 1;
         try {
-            while ((line = bufferedReader.readLine()) != null) {
+            while (true) {
```

Comment:
this is a really gnarly nesting of activities `try -> while -> try -> catch -> if/else -> if` with many routes through it and is difficult to follow. I think it should be broken down into multiple methods. Maybe using the common `withRetry(int times, Callable<...> c)` pattern we use in other plugins.

Reply:
Updated